File: Python/maxx.py
# find maximum value in a list
def find_max(nums):
    return max(nums)

import time

# choose n to get t = 0.1 ... 1.0 sec
n = 40_000_000
nums = n * [1]  # best case
# nums = range(n) # worst case
t = time.perf_counter()
x_max = find_max(nums)
dt = time.perf_counter() - t
print(n, dt)
        
# (c) Valentin Arkov

File: CS/maxx.cs
using System;
using System.Diagnostics;
using System.Linq;

class Program
{
    // find maximum in array (built-in)
    static int FindMax(int[] nums)
    {
        return nums.Max();
    }

    static void Main(string[] args)
    {
        // Choose array size (0.1 ... 1 sec)
        int n = 1_000_000_000;
        int[] nums = new int[n];
        for (int i = 0; i < n; i++)
            nums[i] = 1;   // Best case
            // nums[i] = i;      // Worst case

        Stopwatch stopwatch = Stopwatch.StartNew();
        int x_max = FindMax(nums);
        stopwatch.Stop();
        double t = stopwatch.Elapsed.TotalSeconds;

        Console.WriteLine($"{n} {t}");
    }
}

// (c) Valentin Arkov